### ISSUE TITLE
svn: E125002: '/trunk@555' does not appear to be a URL  

### ISSUE THREAD

**Issue by markingman** [composer/composer#5842]
I'm trying to install a local SVN repository but keep getting "svn: E125002" error.

Appears the result svn co command might not prefix the file:/// path?

My `composer.json` is like:

```json
{
    "minimum-stability": "dev",
    "autoload": {
        "psr-0": {
            "App\\": "app/classes"
        },
        "psr-4": {
            "App\\": "app/classes"
        }
    },
    "require-dev": {
        "phpunit/phpunit": "^5.6"
    },
    "require": {
        "php": ">=5.6"
    },
	"repositories": [
        {
            "type": "vcs",
            "url": "/Full/path/to/local/svn/repo"
        }
    ]
}
```

Output of `composer diagnose` is like:

```
Dependency resolution completed in 0.004 seconds
  - Installing test/test (dev-trunk)
    Checking out /trunk/@555
Failed: [RuntimeException] Package could not be downloaded, 
svn: E125002: '/trunk@555' does not appear to be a URL


Installation failed, reverting ./composer.json to its original content.

                                                          
  [RuntimeException]                                      
  Package could not be downloaded,                        
  svn: E125002: '/trunk@589' does not appear to be a URL  
                                                          

Exception trace:
 () at phar:///Users/mark/project/composer.phar/src/Composer/Downloader/SvnDownloader.php:103
 Composer\Downloader\SvnDownloader->execute() at phar:///Users/mark/project/composer.phar/src/Composer/Downloader/SvnDownloader.php:44
 Composer\Downloader\SvnDownloader->doDownload() at phar:///Users/mark/project/composer.phar/src/Composer/Downloader/VcsDownloader.php:72
 Composer\Downloader\VcsDownloader->download() at phar:///Users/mark/project/composer.phar/src/Composer/Downloader/DownloadManager.php:213
 Composer\Downloader\DownloadManager->download() at phar:///Users/mark/project/composer.phar/src/Composer/Installer/LibraryInstaller.php:186
 Composer\Installer\LibraryInstaller->installCode() at phar:///Users/mark/project/composer.phar/src/Composer/Installer/LibraryInstaller.php:90
 Composer\Installer\LibraryInstaller->install() at phar:///Users/mark/project/composer.phar/src/Composer/Installer/InstallationManager.php:173
 Composer\Installer\InstallationManager->install() at phar:///Users/mark/project/composer.phar/src/Composer/Installer/InstallationManager.php:160
 Composer\Installer\InstallationManager->execute() at phar:///Users/mark/project/composer.phar/src/Composer/Installer.php:542
 Composer\Installer->doInstall() at phar:///Users/mark/project/composer.phar/src/Composer/Installer.php:216
 Composer\Installer->run() at phar:///Users/mark/project/composer.phar/src/Composer/Command/RequireCommand.php:175
 Composer\Command\RequireCommand->execute() at phar:///Users/mark/project/composer.phar/vendor/symfony/console/Command/Command.php:259
 Symfony\Component\Console\Command\Command->run() at phar:///Users/mark/project/composer.phar/vendor/symfony/console/Application.php:847
 Symfony\Component\Console\Application->doRunCommand() at phar:///Users/mark/project/composer.phar/vendor/symfony/console/Application.php:192
 Symfony\Component\Console\Application->doRun() at phar:///Users/mark/project/composer.phar/src/Composer/Console/Application.php:231
 Composer\Console\Application->doRun() at phar:///Users/mark/project/composer.phar/vendor/symfony/console/Application.php:123
 Symfony\Component\Console\Application->run() at phar:///Users/mark/project/composer.phar/src/Composer/Console/Application.php:104
 Composer\Console\Application->run() at phar:///Users/mark/project/composer.phar/bin/composer:43
 require() at /Users/mark/project/composer.phar:24
```

When I run this command:

```
php composer.phar require -vvv "test/test @dev"
```

Is there any way to run Composer as source or something — keen to see more info from SvnDownloader->execute()

Have tried various combinations — but seems if it finds a .json file, it attempts svn co but calls the path and not full file:/// URL.

cheers,
Mark

**Comment by stof** [comment]
to run composer from source, here is the way:

```bash
$ git clone https://github.com/composer/composer.git ~/project/composer
$ cd ~/project/composer
$ git checkout 1.2 # use the 1.2 version of composer rather than the master branch (optional)
$ composer install # yes, composer relies on itself for its dependencies
# at this point, you have a valid local setup of composer
$ cd ~/project/other # go to the project where you want to use composer
$ ../composer/bin/composer update # adapt the path depending on the location you used
```

**Comment by markingman** [comment]
Thanks very much for that — I got Composer running from source write away.

I'll have a deeper look shortly but noted for some reason SvnDownloader->doDownload(...) is called with no $url... When I hacked in my 'file:///...' URL path as the $url value (same as I'd set in composer.json) it seems to have worked (it went on to do the svn co correctly).

On the right track at least — I'll update when I can see how/why it didn't set the $url var.

Thanks again stof! 👍
